### PR TITLE
Add ck_assert_pstr_* macros with ability to compare undefined (NULL) strings

### PR DESCRIPTION
**Pull request by 9fcc**
I decided to open another merge request for this feature to simplify code review. So fist ck_assert_str_* fixes could be merged and then these.

### CODE REVIEW

**Comment by brarcher** [comment]
Jenkins: ok to test

File: doc/check.texi [hunk 1]
```diff
@@ -1204,9 +1204,21 @@ operator is again indicated by last two letters of the function name.
...
 
+@item ck_assert_str_eq
+@itemx ck_assert_str_ne
+
+
+
 @item ck_assert_ptr_eq
 @itemx ck_assert_ptr_ne
 
+Similar to @code{ck_assert_str_*} macros, but able to check undefined strings.
```
Comment (brarcher) [inline doc/check.texi]: Did you intend this section to be before the ck_assert_ptr_* macros?

File: doc/check.texi [hunk 2]
```diff
@@ -1204,9 +1204,21 @@ operator is again indicated by last two letters of the function name.
 of the character string @code{a} is less than that of @code{b}.
 If a NULL pointer would be passed to any comparison macro than check would fail.
 
+@item ck_assert_str_eq
```
Comment (brarcher) [inline doc/check.texi]: Did you mean ck_asserts_**p**str_eq

File: src/check.h.in [hunk 3]
```diff
@@ -1487,6 +1487,46 @@ do { \
...
+ * @param X string
+ * @param Y string to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.11.0
+ */
+#define ck_assert_pstr_ne(X, Y) _ck_assert_str(X, !=, Y, 0, 1)
+
+/* Memory location comparison macros with improved output compared to ck_assert() */
```
Comment (brarcher) [inline src/check.h.in]: This looks like a duplicate of what comes after it. Did you mean to include it?

Comment (9fcc) [inline src/check.h.in]: Do you mean that `ck_assert_pstr_ne` does the same as `ck_assert_str_ne` does? If so, no, `ck_assert_str_ne` always fails if `NULL` passed.

**Comment by brarcher** [comment]
Thanks for the contribution!

**Pull request merged**